game/game/api/shortcuts.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+import game.constants
+import game.packets
+from common.api_handlers import l2_request_handler
+from common.ctype import ctype
+from common.template import Parameter, Template
+from game.models import Character
+from game.models.structures.shortcut import Shortcut
+from game.request import GameRequest
+
+LOG = logging.getLogger(f"l2py.{__name__}")
+
+
+@l2_request_handler(
+    game.constants.GAME_REQUEST_SHORTCUT_REG,
+    Template(
+        [
+            Parameter(id="type", start=0, length=4, type=ctype.int32),
+            Parameter(id="slot", start="$type.stop", length=4, type=ctype.int32),
+            Parameter(id="id", start="$slot.stop", length=4, type=ctype.int32),
+        ]
+    ),
+    states="*",
+)
+async def request_short_cut_reg(request: GameRequest):
+    LOG.debug("saving shortcut", request)
+    character: Character = request.session.character
+    shortcut = Shortcut(
+        slot=ctype.int32(request.validated_data["slot"].value % 12),
+        page=ctype.int32(request.validated_data["slot"].value / 12),
```

Comment:
is access to `.value` nescessary?

Reply:
need to add `__imod__` to `ctype._Numeric`

Reply:
> it's necessary.



> need to add to `ctype._Numeric`
> 
> ```
>     def __imod__(self, other):
>         other_value = other
>         if isinstance(other, _Numeric):
>             other_value = other.value
>         self.value = self.value % other_value
>         return self
> 
>     def __mod__(self, other):
>         other_value = other
>         if isinstance(other, _Numeric):
>             other_value = other.value
>         return self.__class__(self.value % other_value)
> ```

I tried this, doesn't work

Reply:
It works!